File: javascript/gemini/p103.js
const fs = require('fs');

// 'w' flag in Node corresponds to O_TRUNC | O_CREAT | O_WRONLY
fs.open('example.txt', 'w', (err, fd) => {
    if (err) throw err;
    
    fs.write(fd, 'Hello JS', (err) => {
        if (err) throw err;
        fs.close(fd, () => console.log('Done!'));
    });
});

File: c/gemini/p103.c
#include <fcntl.h>
#include <stdio.h>
#include <unistd.h>

int main() {
    // O_CREAT: Create file
    // O_WRONLY: Write only
    // O_TRUNC: Clear existing content
    int fd = open("example.txt", O_WRONLY | O_CREAT | O_TRUNC, 0644);
    
    if (fd == -1) {
        perror("Error opening file");
        return 1;
    }
    
    write(fd, "Hello C\n", 8);
    close(fd);
    return 0;
}